game/plugins/src/main/kotlin/gg/rsmod/plugins/content/items/ringofdueling/ring_of_dueling.plugin.kts

```diff
@@ -0,0 +1,108 @@
+package gg.rsmod.plugins.content.items.ringofdueling
+
+import gg.rsmod.plugins.content.magic.TeleportType
+import gg.rsmod.plugins.content.magic.canTeleport
+
+val RING_OF_DUELING = intArrayOf(
+        Items.RING_OF_DUELING8, Items.RING_OF_DUELING7, Items.RING_OF_DUELING6,
+        Items.RING_OF_DUELING5, Items.RING_OF_DUELING4, Items.RING_OF_DUELING3,
+        Items.RING_OF_DUELING2, Items.RING_OF_DUELING1
+)
+
+RING_OF_DUELING.forEach { duel ->
+    on_equipment_option(duel, option = "Duel Arena") {
+        player.queue(TaskPriority.STRONG) {
+            teleport(Tile(x = 3308, z = 3234, height = 0)) //Coordinates of duel arena
+        }
+    }
+
+    on_equipment_option(duel, option = "Castle Wars") {
+        player.queue(TaskPriority.STRONG) {
+            teleport(Tile(x = 2440, z = 3089, height = 0)) //Coordinates of castle wars
+        }
+    }
+
+    on_equipment_option(duel, option = "Clan Wars") {
+        player.queue(TaskPriority.STRONG) {
+            teleport(Tile(x = 3370, z = 3161, height = 0)) //Coordinates of clan wars
+        }
+    }
+}
+
+fun QueueTask.teleport(tile : Tile) {
+    if (player.canTeleport(TeleportType.MODERN)) {
+        if (player.hasEquipped(EquipmentType.RING, *RING_OF_DUELING)) {
+            world.spawn(AreaSound(tile, id = 200, radius = 10, volume = 1))
```

Comment:
The `AreaSound` should be a constant instead to avoid [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)).